amazon-kinesis-client/src/main/java/software/amazon/kinesis/retrieval/fanout/FanOutRetrievalFactory.java

```diff
@@ -41,36 +41,23 @@ public class FanOutRetrievalFactory implements RetrievalFactory {
     private final String defaultConsumerArn;
     private final Function<String, String> consumerArnCreator;
 
-    private Map<StreamIdentifier, String> implicitConsumerArnTracker = new HashMap<>();
-
-    @Override
-    public GetRecordsRetrievalStrategy createGetRecordsRetrievalStrategy(final ShardInfo shardInfo,
-            final MetricsFactory metricsFactory) {
-        return null;
-    }
+    private final Map<StreamIdentifier, String> implicitConsumerArnTracker = new HashMap<>();
 
     @Override
     public RecordsPublisher createGetRecordsCache(@NonNull final ShardInfo shardInfo,
-            final StreamConfig streamConfig,
-            final MetricsFactory metricsFactory) {
+            @NonNull final StreamConfig streamConfig,
+            @Nullable final MetricsFactory metricsFactory) {
```

Comment:
This is not even used?

Reply:
Correct, the `MetricsFactory` argument doesn't seem to actually be used in the `FanOutRetrievalFactory` implementation.